pkg/kubelet/server/server_test.go

```diff
@@ -546,6 +556,9 @@ func TestAuthzCoverage(t *testing.T) {
 }
 
 func TestAuthFilters(t *testing.T) {
+	// Enable features.ContainerCheckpoint during test
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ContainerCheckpoint, true)()
```

Comment:
Can we also test this with the feature flag enabled? Does any behaviour change?

Reply:
Do you mean with the feature flag **disabled**? Maybe I am using `SetFeatureGateDuringTest()` wrongly. I am currently confused why it is using `defer`.

I am running the tests like this: `KUBE_TEST_ARGS="-v -failfast" bash -x hack/make-rules/test.sh ./pkg/kubelet/server` and without the defer line I get following error:
```
=== RUN   TestAuthFilters/POST:/checkpoint/{podNamespace}/{podName}/{containerName}
    server_test.go:599: 
        	Error Trace:	server_test.go:599
        	Error:      	Not equal: 
        	            	expected: 403
        	            	actual  : 404
        	Test:       	TestAuthFilters/POST:/checkpoint/{podNamespace}/{podName}/{containerName}
    server_test.go:600: 
        	Error Trace:	server_test.go:600
        	Error:      	Should be true
        	Test:       	TestAuthFilters/POST:/checkpoint/{podNamespace}/{podName}/{containerName}
        	Messages:   	Authenticate was not called
    server_test.go:601: 
        	Error Trace:	server_test.go:601
        	Error:      	Should be true
        	Test:       	TestAuthFilters/POST:/checkpoint/{podNamespace}/{podName}/{containerName}
        	Messages:   	Attributes were not called
    server_test.go:602: 
        	Error Trace:	server_test.go:602
        	Error:      	Should be true
        	Test:       	TestAuthFilters/POST:/checkpoint/{podNamespace}/{podName}/{containerName}
        	Messages:   	Authorize was not called
--- FAIL: TestAuthFilters (0.26s)
```
With the defer line the test passes. My idea was to enable the feature during this test to make the test pass, but I am no longer sure that makes sense. Any recommendations from your side how to use it correctly?

Reply:
The current behaviour here is correct - featuregatetesting.SetFeatureGateDuringTest(...) will immediately set the feature gate, and returns a function to revert the feature gate to its original value. This is the function that is then `defer`'d